skiko/src/awtMain/kotlin/org/jetbrains/skiko/PlatformOperations.kt

```diff
@@ -6,34 +6,48 @@ import java.awt.event.ComponentAdapter
 import java.awt.event.ComponentEvent
 import javax.swing.SwingUtilities
 
-internal open class FullscreenAdapter(
+internal class FullscreenAdapter(
     val backedLayer: HardwareLayer
 ): ComponentAdapter() {
-    private var _isFullscreenDispatched = false
-    private var _isFullscreen: Boolean = false
+
+    // Keep a `localFullscreen` flag which stores the virtual fullscreen state when the window is not actually
+    // visible, and apply it when the window does become visible.
+    private var localFullscreen = false
+
+    // Additionally, keep an `isWindowShown` flag that says whether `backedLayer` or `localFullscreen` is currently
+    // the source of truth. This flag must be used, rather than the real state (e.g. with `window.isVisible)
+    // because otherwise the code becomes dependent on the order of listener calls. For example, `componentResized`
+    // can be called when the window is already visible but before `componentShown` has been called. If the test
+    // in `componentResized` was on `window.isVisible`, it would reset the value of `localFullscreen` before it had
+    // applied in `componentShown`.
+    private var isWindowShown = false
```

Comment:
It might work, but it's safer to rely on the flag that's under our control.
If, for some reason `SwingUtilities.getRoot(backedLayer).isVisible` returned `false` inside `componentResized` it would again break the code.